### ISSUE TITLE
Accessibility: "Follow us" header tag in footer breaks Lighthouse audits.

### ISSUE THREAD

**Issue by caroqliu** [ampproject/amp.dev#4431]
## 🐛 Bug Report
"Follow us" header tag in site-level footer breaks Lighthouse audits.
![image](https://user-images.githubusercontent.com/10456171/92026493-1d5b7400-ed2f-11ea-986b-4f18b0b5c105.png)

### Current behavior
Failing lighthouse accessibility audit: `Heading elements are not in a sequentially-descending order`
[Relevant resource](http://web.dev/heading-order/?utm_source=lighthouse)

### Expected behavior/code
Appropriate heading hierarchy identifying the section. We can fix this by replacing `<h5>` with `<header>`, making sure styles also transfer as needed.

### Reproduction procedure
1. Go to amp.dev 
2. Open Lighthouse in Chrome DevTools
3. Run an Accessibility audit
4. See findings
